### PR TITLE
Refactor iiif canvas and manifest transformers

### PR DESCRIPTION
**Pull request by gestchild**
For #10606

I think the issue largely explains what this PR is doing

### CODE REVIEW

File: content/webapp/components/IIIFViewer/ViewerStructures.tsx [hunk 1]
```diff
@@ -74,15 +72,17 @@ const Structures: FunctionComponent<Props> = ({
   work,
   query,
 }) => {
-  return ranges.length > 0 ? (
+  return ranges && ranges.length > 0 ? (
```
Comment (davidpmccormick) [inline content/webapp/components/IIIFViewer/ViewerStructures.tsx:75]: Is optional chaining (`ranges?.length > 0`) no good here for TypeScript reasons or something?

File: content/webapp/test/utils/iiif.test.ts [hunk 2]
```diff
@@ -165,7 +158,7 @@ const correctResult = [
 ];
 
 describe('Group repetitive iiif ranges', () => {
-  it('groups iiif ranges with consecutive canvases and the same label', () => {
+  it.only('groups iiif ranges with consecutive canvases and the same label', () => {
```
Comment (davidpmccormick) [inline content/webapp/test/utils/iiif.test.ts:161]: Can remove this now?

**Review by davidpmccormick — APPROVED** [review]
💪

File: content/webapp/components/IIIFViewer/ViewerStructures.tsx (hunk 1 above)
Comment (gestchild) [reply content/webapp/components/IIIFViewer/ViewerStructures.tsx:75]: yeah, it complains ranges.length could be undefined

**Pull request merged**